apps/web/src/actions/copilot/refinePrompt.ts

```diff
@@ -0,0 +1,87 @@
+'use server'
+
+import { BadRequestError } from '@latitude-data/core/lib/errors'
+import {
+  DocumentVersionsRepository,
+  EvaluationResultsRepository,
+  EvaluationsRepository,
+} from '@latitude-data/core/repositories'
+import { serialize as serializeEvaluationResult } from '@latitude-data/core/services/evaluationResults/serialize'
+import { env } from '@latitude-data/env'
+import { createSdk } from '$/app/(private)/_lib/createSdk'
+import { z } from 'zod'
+
+import { authProcedure } from '../procedures'
+
+export const refinePromptAction = authProcedure
+  .createServerAction()
+  .input(
+    z.object({
+      projectId: z.number(),
+      commitUuid: z.string(),
+      documentUuid: z.string(),
+      evaluationId: z.number(),
+      evaluationResultIds: z.array(z.number()),
+    }),
+  )
+  .handler(async ({ ctx, input }) => {
+    if (!env.COPILOT_WORKSPACE_APIKEY) {
+      throw new BadRequestError('COPILOT_WORKSPACE_APIKEY is not set')
+    }
+    if (!env.COPILOT_PROJECT_ID) {
+      throw new BadRequestError('COPILOT_PROJECT_ID is not set')
+    }
+    if (!env.COPILOT_REFINE_PROMPT_PATH) {
+      throw new BadRequestError('COPILOT_REFINE_PROMPT_PATH is not set')
+    }
```

Comment:
Not to change now. But is a pain to have to define all these checks. I think we should make all mandatory and let our env take care of it on runtime. By [setting it here](https://github.com/latitude-dev/latitude-llm/blob/main/apps/web/src/env.ts#L14)